### PR TITLE
LPD-39407 Updated the Installation flow, and installation footer buttons

### PR DESCRIPTION
**Pull request by welitonpe**

### CODE REVIEW

**Comment by welitonpe** [comment]
ci:test:sf

**Comment by welitonpe** [comment]
ci:test:sf

**Review by eduardo-diniz — APPROVED** [review]
LGTM :fire:

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 1]
```diff
@@ -19,11 +22,17 @@ type EnvironmentRadioProps = {
 	setValue: UseFormSetValue<z.infer<typeof zodSchema.installProductSchema>>;
 };
 
+const CUSTOMFIELD_KEY = 'cloud-provisioning';
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:25]: Use `ORDER_CUSTOM_FIELDS.CLOUD_PROVISIONING` instead

```suggestion
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 2]
```diff
@@ -19,11 +22,17 @@ type EnvironmentRadioProps = {
...
+
 const EnvironmentRadio: React.FC<EnvironmentRadioProps> = ({
 	selectedEnvironment,
 	selectedProject,
 	setValue,
 }) => {
+	const {placedOrder} = useOutletContext<CloudProvisioningOutletContext>();
+	const deploiments = JSON.parse(placedOrder.customFields[CUSTOMFIELD_KEY])[0]
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:33]:
```suggestion
	const deployments = JSON.parse(placedOrder.customFields[CUSTOMFIELD_KEY])[0]
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 3]
```diff
@@ -34,12 +43,20 @@ const EnvironmentRadio: React.FC<EnvironmentRadioProps> = ({
 				const [projectName = '', environment = ''] =
 					projectEnvironment.projectId.split('-');
 
+				const hasDisabled = deploiments.find((deployment: any) => {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:46]:
```suggestion
				const disabled = deployments.find((deployment: any) => {
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 4]
```diff
@@ -34,12 +43,20 @@ const EnvironmentRadio: React.FC<EnvironmentRadioProps> = ({
...
+					const [_, hasInstallationOnEnvironment = ''] =
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:47]:
```suggestion
					const [_, installionEnvironment = ''] =
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/pages/CloudProvisioningOutlet.tsx [hunk 5]
```diff
@@ -205,6 +206,7 @@ export type CloudProvisioningOutletContext = {
 	onClickCancel: () => void;
 	onSubmit: () => void;
 	orderId: number;
+	placedOrder: any;
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/pages/CloudProvisioningOutlet.tsx:209]:
```suggestion
	placedOrder: PlacedOrder;
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by welitonpe** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 6]
```diff
@@ -24,22 +29,40 @@ const EnvironmentRadio: React.FC<EnvironmentRadioProps> = ({
...
+		}>
+	) => setValue('environment', selectedRadio.value);
 
 	return (
 		<>
 			{selectedProject?.environments?.map((projectEnvironment, index) => {
 				const [projectName = '', environment = ''] =
 					projectEnvironment.projectId.split('-');
 
+				const disabled = provisions.find((provisioning: Provisions) =>
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:50]:
```suggestion
				const disabled = provisions.find((provisioning: Provisioning) =>
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 7]
```diff
@@ -5,11 +5,16 @@
...
+import {useOutletContext} from 'react-router-dom';
 import {z} from 'zod';
 
 import RadioCard from '../../../../../../../components/RadioCardList/components/RadioCard';
+import {ORDER_CUSTOM_FIELDS} from '../../../../../../../enums/Order';
+import i18n from '../../../../../../../i18n';
 import zodSchema from '../../../../../../../schema/zod';
 import {ConsoleUserProject} from '../../../../../../../services/oauth/types';
+import {CloudProvisioningOutletContext} from '../pages/CloudProvisioningOutlet';
+import {Deployment, Provisions} from '../types';
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:17]:
```suggestion
import {Deployment, Provisioning} from '../types';
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx [hunk 8]
```diff
@@ -24,22 +29,40 @@ const EnvironmentRadio: React.FC<EnvironmentRadioProps> = ({
 	selectedProject,
 	setValue,
 }) => {
-	const handleSelectRadio = (selectedRadio: RadioOption<any>) => {
-		setValue('environment', selectedRadio.value);
-	};
+	const {placedOrder} = useOutletContext<CloudProvisioningOutletContext>();
+	const provisions = JSON.parse(
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/components/EnvironmentRadio.tsx:33]:
```suggestion
	const provisioning = JSON.parse(
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/types/index.ts [hunk 9]
```diff
@@ -37,6 +37,22 @@ export type StepsInformationProps = {
 	subTitle: any;
 };
 
+export type Provisions = {
```
Comment (kevenleone) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/CustomerDashboard/pages/Apps/App/CloudProvisioning/types/index.ts:40]:
```suggestion
export type Provisioning = {
```

**Review by kevenleone — CHANGES_REQUESTED** [review]

**Comment by welitonpe** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**